In [14]:
import sys
import os

# Get absolute path to project root (one level above notebooks/)
project_root = os.path.abspath(os.path.join(os.getcwd(), ".."))

# Add to Python path if not already added
if project_root not in sys.path:
    sys.path.append(project_root)

print("Python Path Updated!")
print("Project root:", project_root)


Python Path Updated!
Project root: c:\Users\Josh\Desktop\kaim-week3


In [15]:
from src.hypothesis_tests import (
    load_data,
    create_bmi_category,
    t_test_groups,
    anova_test,
    chi_square_test,
    interpret_p_value
)

df = load_data("../data/insurance.csv")
df.head()


Loaded dataset successfully: 1338 rows, 7 columns


age     sex     bmi  children smoker     region      charges
0   19  female  27.900         0    yes  southwest  16884.92400
1   18    male  33.770         1     no  southeast   1725.55230
2   28    male  33.000         3     no  southeast   4449.46200
3   33    male  22.705         0     no  northwest  21984.47061
4   32    male  28.880         0     no  northwest   3866.85520

In [16]:
# Hypothesis 1: Smoker vs Non-Smoker

smokers = df[df["smoker"] == "yes"]["charges"]
nonsmokers = df[df["smoker"] == "no"]["charges"]

t_stat, p_val = t_test_groups(smokers, nonsmokers)

print("T-test: Smoker vs Non-Smoker")
print("t-statistic:", t_stat)
print("p-value:", p_val)
print(interpret_p_value(p_val))


T-test: Smoker vs Non-Smoker
t-statistic: 32.751887766341824
p-value: 5.88946444671698e-103
Reject H₀ (p = 0.0000)


In [8]:
# Hypothesis 2: Male vs Female

male = df[df["sex"] == "male"]["charges"]
female = df[df["sex"] == "female"]["charges"]

t_stat, p_val = t_test_groups(male, female)

print("T-test: Male vs Female")
print("t-statistic:", t_stat)
print("p-value:", p_val)
print(interpret_p_value(p_val))


T-test: Male vs Female
t-statistic: 2.1008878232359565
p-value: 0.03584101495601664
Reject H₀ (p = 0.0358)


In [9]:
# Hypothesis 3: Region effect on charges (ANOVA)

f_stat, p_val = anova_test(df, "region")

print("ANOVA: Charges across Regions")
print("F-statistic:", f_stat)
print("p-value:", p_val)
print(interpret_p_value(p_val))


ANOVA: Charges across Regions
F-statistic: 2.96962669358912
p-value: 0.0308933560705201
Reject H₀ (p = 0.0309)


In [12]:
df = create_bmi_category(df)
df.head()


age     sex     bmi  children smoker     region      charges   bmi_group
0   19  female  27.900         0    yes  southwest  16884.92400  normal_bmi
1   18    male  33.770         1     no  southeast   1725.55230    high_bmi
2   28    male  33.000         3     no  southeast   4449.46200    high_bmi
3   33    male  22.705         0     no  northwest  21984.47061  normal_bmi
4   32    male  28.880         0     no  northwest   3866.85520  normal_bmi

In [13]:
# Hypothesis 4: BMI Group (High BMI vs Normal BMI)

high_bmi = df[df["bmi_group"] == "high_bmi"]["charges"]
normal_bmi = df[df["bmi_group"] == "normal_bmi"]["charges"]

t_stat, p_val = t_test_groups(high_bmi, normal_bmi)

print("T-test: High BMI vs Normal BMI")
print("t-statistic:", t_stat)
print("p-value:", p_val)
print(interpret_p_value(p_val))


T-test: High BMI vs Normal BMI
t-statistic: 7.679158554353561
p-value: 3.5040049055619615e-14
Reject H₀ (p = 0.0000)


🔍 Interpretation — Smoker vs Non-Smoker

H₀: No difference in charges between smokers and non-smokers.

H₁: Smokers have different charges.

Result: p-value ≈ 0
👉 Reject H₀

Meaning:
Smokers pay much higher charges. This is the strongest risk factor in the dataset.

Business Impact:
Smoking status should be used for premium differentiation — smokers represent high-risk clients.

🔍 Interpretation — Male vs Female

H₀: No difference in mean charges.

H₁: A difference exists.

Result: p-value = 0.0358
👉 Reject H₀, but the effect is small.

Meaning:
There is a small difference in charges between genders, but not strong enough to drive premium changes alone.

Business Impact:
Gender should not be a major premium factor — consider it only when combined with BMI or smoking.


Interpretation — Region Differences (ANOVA)

H₀: All regions have equal mean charges.

H₁: At least one region differs.

Result: p-value = 0.0309
👉 Reject H₀

Meaning:
Charges vary between regions. Possibly due to healthcare cost differences.

Business Impact:
Insurance pricing might need regional adjustments

🔍 Interpretation — BMI Category (High BMI vs Normal BMI)

H₀: No difference in mean charges.

H₁: Charges differ.

Result: p-value ≈ 0
👉 Reject H₀

Meaning:
High BMI individuals pay significantly more. Obesity is a major risk factor.

Business Impact:
BMI should be included in risk scoring.
Wellness programs could reduce long-term costs.

📊 Summary of Hypothesis Testing

All four tests found statistically significant differences:

Factor Tested	Significant?	Strength of Effect
Smoking	✔ Yes	⭐⭐⭐⭐⭐ Very strong
BMI	✔ Yes	⭐⭐⭐⭐ Strong
Region	✔ Yes	⭐⭐ Moderate
Gender	✔ Yes	⭐ Low
Final Conclusion

Smoking and BMI are the two strongest predictors of high medical charges.
These factors should guide premium optimization and customer risk segmentation.